pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -106,6 +123,7 @@ func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageClust
 				"csi.storage.k8s.io/node-stage-secret-namespace":  initData.Namespace,
 			},
 		},
+		r.newOBCStorageClass(initData),
```

Comment:
My understanding is that we do not want to create the OBC StorageClass in all scenarios. There are only two scenarios we want to deploy it in:

1. External StorageClusters that have an RGW endpoint defined in their JSON blob
2. Converged StorageClusters that are not in a cloud environment, similar to the behavior in `ensureCephObjectStores()`

Reply:
Let me double-check for 2.

Reply:
Ok looks like there is a mechanism in place already whether we should deploy CephObjectStore, so we should create the SC only if that's the case.

Reply:
@leseb , @jarrpa , made the change accordingly...